internal/injector/builtin/yaml/api/vault.yml

```
@@ -0,0 +1,34 @@
+# Unless explicitly stated otherwise all files in this repository are licensed
+# under the Apache License Version 2.0.
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2024-present Datadog, Inc.
+%YAML 1.1
+---
+meta:
+  name: github.com/hashicorp/vault/api
+  description: Official package for interacting with a Vault server.
+  icon: at-symbol
```

(Optional) I think it might be a nice touch to use a shield or key icon instead for this one:

```suggestion
  icon: shield-check
```

```suggestion
  icon: key
```